### ISSUE TITLE
SIGABRT and SIGSEGV during a frequent reload (main_loop_reload_config_apply)

### ISSUE THREAD

**Issue by mitzkia** [balabit/syslog-ng#1349]
# syslog-ng

## Version of syslog-ng

```
OSE master: 3.9.1
builded from source
last commit: 4b0b39971949dbc7844d61a232b2b6b370e01510
```

## Platform

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.10
Release:	16.10
Codename:	yakkety

$ uname -a
Linux tristram 4.8.0-37-generic #39-Ubuntu SMP Thu Jan 26 02:27:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

# Issue

## Failure

I know that this will be not the usual working mode for syslog-ng (but the crash happened). 
As I saw the relevant code has changed recently and it would be good to check it again.

During the reproduction I got some different backtraces.

1, start syslog-ng with the following minimal config :
- I don't know why but mongodb() driver can trigger this issue

```
@version: 3.9
source s_internal { internal();};source s_file {file('/tmp/inputf');};
destination d_mongodb {mongodb();};
log {
	source(s_internal);	source(s_file);
	destination(d_mongodb);
};
```
2, Use the following command line to trigger crashes:

```
while true ; do pkill -1 syslog-ng ;  echo  "test message" >> /tmp/inputf ; sbin/syslog-ng-ctl stats ;  done
```

I got the following different backtraces (for different runs, but using the same method)

bt1:

```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
#1  0x00007ffff72353ea in __GI_abort () at abort.c:89
#2  0x00007ffff72770d0 in __libc_message (do_abort=do_abort@entry=2, fmt=fmt@entry=0x7ffff738c368 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff727f128 in malloc_printerr (ar_ptr=0x7ffff75bfb00 <main_arena>, ptr=<optimized out>, str=0x7ffff7388ede "corrupted double-linked list", action=3) at malloc.c:5046
#4  malloc_consolidate (av=av@entry=0x7ffff75bfb00 <main_arena>) at malloc.c:4218
#5  0x00007ffff7281d7f in _int_malloc (av=av@entry=0x7ffff75bfb00 <main_arena>, bytes=bytes@entry=4096) at malloc.c:3485
#6  0x00007ffff7283de4 in __GI___libc_malloc (bytes=4096) at malloc.c:2925
#7  0x00007ffff726c382 in __GI__IO_file_doallocate (fp=0x555555806da0) at filedoalloc.c:101
#8  0x00007ffff727b876 in __GI__IO_doallocbuf (fp=fp@entry=0x555555806da0) at genops.c:398
#9  0x00007ffff727a764 in _IO_new_file_underflow (fp=0x555555806da0) at fileops.c:564
#10 0x00007ffff727b932 in __GI__IO_default_uflow (fp=0x555555806da0) at genops.c:413
#11 0x00007ffff726df1a in __GI__IO_getline_info (fp=fp@entry=0x555555806da0, buf=buf@entry=0x7fffffffb770 "tristram", n=n@entry=8191, delim=delim@entry=10, extract_delim=extract_delim@entry=1, 
    eof=eof@entry=0x0) at iogetline.c:60
#12 0x00007ffff726e028 in __GI__IO_getline (fp=fp@entry=0x555555806da0, buf=buf@entry=0x7fffffffb770 "tristram", n=n@entry=8191, delim=delim@entry=10, extract_delim=extract_delim@entry=1) at iogetline.c:34
#13 0x00007ffff7277d03 in __GI___fgets_unlocked (buf=buf@entry=0x7fffffffb770 "tristram", n=n@entry=8192, fp=fp@entry=0x555555806da0) at iofgets_u.c:52
#14 0x00007ffff7326793 in __res_vinit (statp=0x7ffff75c3a80 <_res>, preinit=<optimized out>) at res_init.c:196
#15 0x00007ffff7b24401 in app_post_config_loaded () at ../lib/apphook.c:156
#16 0x00007ffff7b3f779 in main_loop_reload_config_apply (user_data=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:241
#17 0x00007ffff7b40c66 in main_loop_worker_job_complete () at ../lib/mainloop-worker.c:246
#18 0x00007ffff7b40517 in main_loop_call_handler (user_data=0x0) at ../lib/mainloop-call.c:113
#19 0x00007ffff7b8b87b in __iv_event_run_pending_events (_tinfo=0x5555557609f0) at ../../../../lib/ivykis/src/iv_event.c:58
#20 0x00007ffff7b8db8f in iv_event_raw_got_event (_this=0x5555557609f8) at ../../../../lib/ivykis/src/iv_event_raw_posix.c:89
#21 0x00007ffff7b8e2fc in iv_fd_poll_and_run (st=0x555555760980, to=0x7fffffffdd60) at ../../../../lib/ivykis/src/iv_fd.c:163
#22 0x00007ffff7b8f2d6 in iv_main () at ../../../../lib/ivykis/src/iv_main_posix.c:117
#23 0x00007ffff7b400f9 in main_loop_run (self=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:519
#24 0x000055555555605b in main (argc=1, argv=0x7fffffffded8) at ../syslog-ng/main.c:300
(gdb) bt full
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
        set = {__val = {0, 7018409457430718054, 8083166874389458992, 3472328296227680288, 2319406834570502192, 2314885530818453552, 2314885530818453536, 2314885530818453536, 8241999043771768864, 
            3991990709698103901, 3975887029566269030, 7162524645506573926, 8104277497642758192, 3472328296227680288, 2319406834570502192, 2314885530818453552}}
        pid = <optimized out>
        tid = <optimized out>
#1  0x00007ffff72353ea in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x3030203030303030, sa_sigaction = 0x3030203030303030}, sa_mask = {__val = {3966017812923691066, 7306077085492471398, 7378697426660503600, 
              3472328529065436774, 3472310978873881120, 3467824696600309808, 2314903123273988656, 2314885530818453536, 2314885530818453536, 2314885530818453536, 746871187794326363, 7378697629483820646, 
              3472328296228087398, 7378697629483820589, 3472329395843524198, 140737488335776}}, sa_flags = 153, sa_restorer = 0x7fffffffb3a0}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x00007ffff72770d0 in __libc_message (do_abort=do_abort@entry=2, fmt=fmt@entry=0x7ffff738c368 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
        ap = {{gp_offset = 40, fp_offset = 32767, overflow_arg_area = 0x7fffffffb3b0, reg_save_area = 0x7fffffffb340}}
        fd = 108
        on_2 = <optimized out>
        list = <optimized out>
        nlist = <optimized out>
        cp = <optimized out>
        written = <optimized out>
#3  0x00007ffff727f128 in malloc_printerr (ar_ptr=0x7ffff75bfb00 <main_arena>, ptr=<optimized out>, str=0x7ffff7388ede "corrupted double-linked list", action=3) at malloc.c:5046
        buf = "00005555557bfeb0"
        cp = <optimized out>
        ar_ptr = 0x7ffff75bfb00 <main_arena>
        ptr = <optimized out>
        str = 0x7ffff7388ede "corrupted double-linked list"
        action = 3
#4  malloc_consolidate (av=av@entry=0x7ffff75bfb00 <main_arena>) at malloc.c:4218
        fb = 0x7ffff75bfb08 <main_arena+8>
        maxfb = 0x7ffff75bfb50 <main_arena+80>
        p = 0x5555557bfe70
        nextp = <optimized out>
        unsorted_bin = 0x7ffff75bfb58 <main_arena+88>
        first_unsorted = <optimized out>
        nextchunk = <optimized out>
        size = 240
        nextsize = <optimized out>
        prevsize = <optimized out>
        nextinuse = <optimized out>
        bck = <optimized out>
        fwd = <optimized out>
#5  0x00007ffff7281d7f in _int_malloc (av=av@entry=0x7ffff75bfb00 <main_arena>, bytes=bytes@entry=4096) at malloc.c:3485
        nb = 4112
        idx = 99
        bin = <optimized out>
        victim = <optimized out>
        size = <optimized out>
        victim_index = <optimized out>
        remainder = <optimized out>
        remainder_size = <optimized out>
        block = <optimized out>
        bit = <optimized out>
        map = <optimized out>
        fwd = <optimized out>
        bck = <optimized out>
        errstr = 0x0
        __func__ = "_int_malloc"
#6  0x00007ffff7283de4 in __GI___libc_malloc (bytes=4096) at malloc.c:2925
        ar_ptr = 0x7ffff75bfb00 <main_arena>
        victim = <optimized out>
---Type <return> to continue, or q <return> to quit---
        hook = <optimized out>
#7  0x00007ffff726c382 in __GI__IO_file_doallocate (fp=0x555555806da0) at filedoalloc.c:101
        size = 4096
        p = <optimized out>
        st = {st_dev = 20, st_ino = 908, st_nlink = 1, st_mode = 33188, st_uid = 0, st_gid = 0, __pad0 = 0, st_rdev = 0, st_size = 172, st_blksize = 4096, st_blocks = 8, st_atim = {tv_sec = 1487267457, 
            tv_nsec = 775213650}, st_mtim = {tv_sec = 1487267457, tv_nsec = 767213650}, st_ctim = {tv_sec = 1487267457, tv_nsec = 771213650}, __glibc_reserved = {0, 0, 0}}
#8  0x00007ffff727b876 in __GI__IO_doallocbuf (fp=fp@entry=0x555555806da0) at genops.c:398
No locals.
#9  0x00007ffff727a764 in _IO_new_file_underflow (fp=0x555555806da0) at fileops.c:564
        count = <optimized out>
#10 0x00007ffff727b932 in __GI__IO_default_uflow (fp=0x555555806da0) at genops.c:413
        ch = <optimized out>
#11 0x00007ffff726df1a in __GI__IO_getline_info (fp=fp@entry=0x555555806da0, buf=buf@entry=0x7fffffffb770 "tristram", n=n@entry=8191, delim=delim@entry=10, extract_delim=extract_delim@entry=1, 
    eof=eof@entry=0x0) at iogetline.c:60
        c = <optimized out>
        len = 0
        ptr = 0x7fffffffb770 "tristram"
#12 0x00007ffff726e028 in __GI__IO_getline (fp=fp@entry=0x555555806da0, buf=buf@entry=0x7fffffffb770 "tristram", n=n@entry=8191, delim=delim@entry=10, extract_delim=extract_delim@entry=1) at iogetline.c:34
No locals.
#13 0x00007ffff7277d03 in __GI___fgets_unlocked (buf=buf@entry=0x7fffffffb770 "tristram", n=n@entry=8192, fp=fp@entry=0x555555806da0) at iofgets_u.c:52
        count = <optimized out>
        result = <optimized out>
        old_error = 0
#14 0x00007ffff7326793 in __res_vinit (statp=0x7ffff75c3a80 <_res>, preinit=<optimized out>) at res_init.c:196
        fp = 0x555555806da0
        cp = <optimized out>
        pp = <optimized out>
        n = <optimized out>
        buf = "tristram\000r 127.0.1.1\n\000 FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN\n\000nf(8)\n\000`\003vUUU\000\000\205\a\200\367\377\177\000\000 \005vUUU\000\000\000:\314H\301\371Pp\000\000\000\000\000\000\000\000\360YUUUU\000\000\070\335\377\377\377\177\000\000\020\030yUUU\000\000(\310\377\377\377\177\000\000\240\273\377\377\377\177\000\000p\271\377\377\221\000\000\000Kڹ\367\377\177\000\000p\334\377\377\377\177\000\000`\270\265\367\377\177\000\000\000\000\000\000\000\000\000\000m"...
        nserv = 0
        have_serv6 = 0
        haveenv = 0
        havesearch = 0
        nsort = 0
        net = <optimized out>
#15 0x00007ffff7b24401 in app_post_config_loaded () at ../lib/apphook.c:156
No locals.
#16 0x00007ffff7b3f779 in main_loop_reload_config_apply (user_data=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:241
        self = 0x7ffff7dd6a80 <main_loop>
        __func__ = "main_loop_reload_config_apply"
#17 0x00007ffff7b40c66 in main_loop_worker_job_complete () at ../lib/mainloop-worker.c:246
No locals.
#18 0x00007ffff7b40517 in main_loop_call_handler (user_data=0x0) at ../lib/mainloop-call.c:113
        site = 0x7ffff37ea5b0
        result = 0x555555760a90
#19 0x00007ffff7b8b87b in __iv_event_run_pending_events (_tinfo=0x5555557609f0) at ../../../../lib/ivykis/src/iv_event.c:58
        ie = 0x7ffff7dd6ee0 <main_task_posted>
        tinfo = 0x5555557609f0
        events = {next = 0x7fffffffd880, prev = 0x7fffffffd880}
#20 0x00007ffff7b8db8f in iv_event_raw_got_event (_this=0x5555557609f8) at ../../../../lib/ivykis/src/iv_event_raw_posix.c:89
        this = 0x5555557609f8
        toread = 8
        buf = "\001\000\000\000\000\000\000\000!\000\000\000\000\000\000\000`\240}UUU\000\000!\000\000\000\000\000\000\000\252\340\267\367\377\177\000\000\205\006\205\367\377\177\000\000\220\267\177UUU\000\000\000:\314H\301\371Ppa\000\000\000\000\000\000\000\260\332\377\377\377\177\000\000\000\000\000\000\000\000\000\000\204\b\205\367\377\177\000\000\060\000\000\000\060\000\000\000 \332\377\377\377\177\000\000P\331\377\377\377\177\000\000\000:\314H\301\371Pp\000\000\000\000\000\000\000\000\240\243}UUU\000\000p\332\377\377\377\177\000\000@\204\200UUU\000\000\340\236\177UUU\000\000a\000\000\000\000\000\000\000\v\000\000\000\000\000\000\000r\365\204\367\377\177\000\000(5\273\367\377\177\000\000"...
        ret = 8
---Type <return> to continue, or q <return> to quit---
#21 0x00007ffff7b8e2fc in iv_fd_poll_and_run (st=0x555555760980, to=0x7fffffffdd60) at ../../../../lib/ivykis/src/iv_fd.c:163
        fd = 0x555555760a08
        active = {next = 0x7fffffffdd20, prev = 0x7fffffffdd20}
#22 0x00007ffff7b8f2d6 in iv_main () at ../../../../lib/ivykis/src/iv_main_posix.c:117
        to = {tv_sec = 0, tv_nsec = 609602021}
        st = 0x555555760980
#23 0x00007ffff7b400f9 in main_loop_run (self=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:519
No locals.
#24 0x000055555555605b in main (argc=1, argv=0x7fffffffded8) at ../syslog-ng/main.c:300
        rc = 0
        ctx = 0x55555575fbb0
        error = 0x0
        main_loop = 0x7ffff7dd6a80 <main_loop>
        exit_before_main_loop_run = 0
```

bt2:

```
(gdb) bt
#0  0x00007ffff7b4a608 in zone_info_free (self=0xffff000000000000) at ../lib/timeutils.c:552
#1  0x00007ffff7b4b833 in time_zone_info_free (self=0x555555800662) at ../lib/timeutils.c:944
#2  0x00007ffff7b75f97 in log_template_options_destroy (options=0x5555557bfea8) at ../lib/template/templates.c:317
#3  0x00007ffff5618355 in _free (d=0x5555557bfc00) at ../modules/afmongodb/afmongodb.c:547
#4  0x00007ffff7b3542e in log_pipe_unref (self=0x5555557bfc00) at ../lib/logpipe.c:88
#5  0x00007ffff7b2a5dc in log_expr_node_free (self=0x555555797610) at ../lib/cfg-tree.c:267
#6  0x00007ffff7b2a59e in log_expr_node_free (self=0x5555557f8780) at ../lib/cfg-tree.c:264
#7  0x00007ffff7b2a59e in log_expr_node_free (self=0x5555557f87f0) at ../lib/cfg-tree.c:264
#8  0x00007ffff781b79a in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#9  0x00007ffff781c5e3 in g_hash_table_remove_all () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#10 0x00007ffff781c64e in g_hash_table_destroy () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#11 0x00007ffff7b2c133 in cfg_tree_free_instance (self=0x5555557bba58) at ../lib/cfg-tree.c:1170
#12 0x00007ffff7b2593c in cfg_free (self=0x5555557bb8c0) at ../lib/cfg.c:482
#13 0x00007ffff7b3f754 in main_loop_reload_config_apply (user_data=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:214
#14 0x00007ffff7b40c66 in main_loop_worker_job_complete () at ../lib/mainloop-worker.c:246
#15 0x00007ffff7b40517 in main_loop_call_handler (user_data=0x0) at ../lib/mainloop-call.c:113
#16 0x00007ffff7b8b87b in __iv_event_run_pending_events (_tinfo=0x5555557609f0) at ../../../../lib/ivykis/src/iv_event.c:58
#17 0x00007ffff7b8db8f in iv_event_raw_got_event (_this=0x5555557609f8) at ../../../../lib/ivykis/src/iv_event_raw_posix.c:89
#18 0x00007ffff7b8e2fc in iv_fd_poll_and_run (st=0x555555760980, to=0x7fffffffdd60) at ../../../../lib/ivykis/src/iv_fd.c:163
#19 0x00007ffff7b8f2d6 in iv_main () at ../../../../lib/ivykis/src/iv_main_posix.c:117
#20 0x00007ffff7b400f9 in main_loop_run (self=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:519
#21 0x000055555555605b in main (argc=1, argv=0x7fffffffded8) at ../syslog-ng/main.c:300
(gdb) bt full
#0  0x00007ffff7b4a608 in zone_info_free (self=0xffff000000000000) at ../lib/timeutils.c:552
No locals.
#1  0x00007ffff7b4b833 in time_zone_info_free (self=0x555555800662) at ../lib/timeutils.c:944
        __func__ = "time_zone_info_free"
#2  0x00007ffff7b75f97 in log_template_options_destroy (options=0x5555557bfea8) at ../lib/template/templates.c:317
        i = 0
#3  0x00007ffff5618355 in _free (d=0x5555557bfc00) at ../modules/afmongodb/afmongodb.c:547
        self = 0x5555557bfc00
#4  0x00007ffff7b3542e in log_pipe_unref (self=0x5555557bfc00) at ../lib/logpipe.c:88
        __func__ = "log_pipe_unref"
#5  0x00007ffff7b2a5dc in log_expr_node_free (self=0x555555797610) at ../lib/cfg-tree.c:267
        next = 0x1
        p = 0x0
#6  0x00007ffff7b2a59e in log_expr_node_free (self=0x5555557f8780) at ../lib/cfg-tree.c:264
        next = 0x0
        p = 0x555555797610
#7  0x00007ffff7b2a59e in log_expr_node_free (self=0x5555557f87f0) at ../lib/cfg-tree.c:264
        next = 0x0
        p = 0x5555557f8780
#8  0x00007ffff781b79a in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#9  0x00007ffff781c5e3 in g_hash_table_remove_all () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#10 0x00007ffff781c64e in g_hash_table_destroy () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#11 0x00007ffff7b2c133 in cfg_tree_free_instance (self=0x5555557bba58) at ../lib/cfg-tree.c:1170
No locals.
#12 0x00007ffff7b2593c in cfg_free (self=0x5555557bb8c0) at ../lib/cfg.c:482
        __func__ = "cfg_free"
#13 0x00007ffff7b3f754 in main_loop_reload_config_apply (user_data=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:214
        self = 0x7ffff7dd6a80 <main_loop>
        __func__ = "main_loop_reload_config_apply"
#14 0x00007ffff7b40c66 in main_loop_worker_job_complete () at ../lib/mainloop-worker.c:246
No locals.
#15 0x00007ffff7b40517 in main_loop_call_handler (user_data=0x0) at ../lib/mainloop-call.c:113
        site = 0x7ffff7fcc5b0
        result = 0x555555760a90
#16 0x00007ffff7b8b87b in __iv_event_run_pending_events (_tinfo=0x5555557609f0) at ../../../../lib/ivykis/src/iv_event.c:58
        ie = 0x7ffff7dd6ee0 <main_task_posted>
        tinfo = 0x5555557609f0
        events = {next = 0x7fffffffd880, prev = 0x7fffffffd880}
#17 0x00007ffff7b8db8f in iv_event_raw_got_event (_this=0x5555557609f8) at ../../../../lib/ivykis/src/iv_event_raw_posix.c:89
        this = 0x5555557609f8
        toread = 8
        buf = "\001\000\000\000\000\000\000\000!\000\000\000\000\000\000\000\000\034\000\344\377\177\000\000!\000\000\000\000\000\000\000\252\340\267\367\377\177\000\000\205\006\205\367\377\177\000\000\220\326\177UUU\000\000\000\003\353g[\026\303\351a\000\000\000\000\000\000\000\260\332\377\377\377\177\000\000\000\000\000\000\000\000\000\000\204\b\205\367\377\177\000\000\060\000\000\000\060\000\000\000 \332\377\377\377\177\000\000P\331\377\377\377\177\000\000\000\003\353g[\026\303\351\000\000\000\000\000\000\000\000\200#\000\344\377\177\000\000p\332\377\377\377\177\000\000 *{UUU\000\000 \201\177UUU\000\000a\000\000\000\000\000\000\000\v\000\000\000\000\000\000\000r\365\204\367\377\177\000\000(5\273\367\377\177\000\000"...
        ret = 8
#18 0x00007ffff7b8e2fc in iv_fd_poll_and_run (st=0x555555760980, to=0x7fffffffdd60) at ../../../../lib/ivykis/src/iv_fd.c:163
        fd = 0x555555760a08
        active = {next = 0x7fffffffdd20, prev = 0x7fffffffdd20}
#19 0x00007ffff7b8f2d6 in iv_main () at ../../../../lib/ivykis/src/iv_main_posix.c:117
        to = {tv_sec = 0, tv_nsec = 678334510}
        st = 0x555555760980
#20 0x00007ffff7b400f9 in main_loop_run (self=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:519
No locals.
#21 0x000055555555605b in main (argc=1, argv=0x7fffffffded8) at ../syslog-ng/main.c:300
        rc = 0
---Type <return> to continue, or q <return> to quit---
        ctx = 0x55555575fbb0
        error = 0x0
        main_loop = 0x7ffff7dd6a80 <main_loop>
        exit_before_main_loop_run = 0
```

bt3:

```
Thread 1 "syslog-ng" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
58	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
#1  0x00007ffff72353ea in __GI_abort () at abort.c:89
#2  0x00007ffff72770d0 in __libc_message (do_abort=do_abort@entry=2, fmt=fmt@entry=0x7ffff738c368 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff7284338 in malloc_printerr (ar_ptr=0x0, ptr=<optimized out>, str=0x7ffff738c390 "munmap_chunk(): invalid pointer", action=<optimized out>) at malloc.c:5046
#4  munmap_chunk (p=<optimized out>) at malloc.c:2854
#5  __GI___libc_free (mem=<optimized out>) at malloc.c:2977
#6  0x00007ffff7b44ed8 in plugin_candidate_free (self=0x5555557b3980) at ../lib/plugin.c:83
#7  0x00007ffff7829aad in g_list_foreach () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#8  0x00007ffff7b45fa0 in plugin_free_candidate_modules (cfg=0x5555557b6360) at ../lib/plugin.c:469
#9  0x00007ffff7b2592a in cfg_free (self=0x5555557b6360) at ../lib/cfg.c:481
#10 0x00007ffff7b3f754 in main_loop_reload_config_apply (user_data=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:214
#11 0x00007ffff7b40c66 in main_loop_worker_job_complete () at ../lib/mainloop-worker.c:246
#12 0x00007ffff7b40517 in main_loop_call_handler (user_data=0x0) at ../lib/mainloop-call.c:113
#13 0x00007ffff7b8b87b in __iv_event_run_pending_events (_tinfo=0x5555557609f0) at ../../../../lib/ivykis/src/iv_event.c:58
#14 0x00007ffff7b8db8f in iv_event_raw_got_event (_this=0x5555557609f8) at ../../../../lib/ivykis/src/iv_event_raw_posix.c:89
#15 0x00007ffff7b8e2fc in iv_fd_poll_and_run (st=0x555555760980, to=0x7fffffffdd60) at ../../../../lib/ivykis/src/iv_fd.c:163
#16 0x00007ffff7b8f2d6 in iv_main () at ../../../../lib/ivykis/src/iv_main_posix.c:117
#17 0x00007ffff7b400f9 in main_loop_run (self=0x7ffff7dd6a80 <main_loop>) at ../lib/mainloop.c:519
#18 0x000055555555605b in main (argc=1, argv=0x7fffffffded8) at ../syslog-ng/main.c:300
```

**Comment by mitzkia** [comment]
Can be fixed by:  #1765

**Comment by mitzkia** [comment]
#1765 was merged. I can not reproduce this issue with the fix.
